CesiumVectorData/include/CesiumVectorData/VectorStyle.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <CesiumUtility/Color.h>
+
+#include <optional>
+
+namespace CesiumVectorData {
+
+/**
+ * @brief The mode used for coloring.
+ */
+enum class ColorMode : uint8_t {
+  /** @brief The normal color mode. The color will be used directly. */
+  Normal = 0,
+  /**
+   * @brief The color will be chosen randomly.
+   *
+   * The color randomization will be applied to each component, with the
+   * resulting value between 0 and the specified color component value. Alpha is
+   * always ignored. For example, if the color was `(R: 0x00, G: 0x77, B: 0x00,
+   * A: 0xFF)`, the resulting randomized value could be `(R: 0x00, G: 0x41, B:
+   * 0x00, A: 0xFF)`, or `(R: 0x00, G: 0x76, B: 0x00, A: 0xFF)`, but never `(R:
+   * 0x00, G: 0xAA, B: 0x00, A: 0xFF)`.
+   */
+  Random = 1
+};
+
+/**
+ * @brief Specifies the color of a style type.
+ */
+struct ColorStyle {
+  /** @brief The color to be used. */
+  CesiumUtility::Color color = CesiumUtility::Color(0xff, 0xff, 0xff, 0xff);
+  /** @brief The color mode to be used. */
+  ColorMode colorMode = ColorMode::Normal;
+
+  /**
+   * @brief Obtains the color specified on this `ColorStyle`.
+   *
+   * For `ColorMode::Normal`, this just returns the value of `color`. For
+   * `ColorMode::Random`, this returns a randomized value obtained based on
+   * the rules described in \ref ColorMode.
+   */
+  CesiumUtility::Color getColor() const;
+};
+
+/** @brief The mode to use when interpreting a given line width. */
+enum class LineWidthMode : uint8_t {
+  /**
+   * @brief The line will always be this number of pixels in width, no matter
+   * how close the user gets to the line.
+   */
+  Pixels = 0,
+  /**
+   * @brief The line width will cover this number of meters of the ellipsoid
+   * it's rendered on. This may cause the line to disappear as the user zooms
+   * out.
+   */
+  Meters = 1
```

Comment:
Clarify that this is meters _at the equator_. These "meters" get smaller as we move toward the equator.

Alternatively, the line width could be expressed directly in radians, and wouldn't need as much explanation. That might be less confusing. Maybe.

Maintaining an actual fixed width in meters would require adjusting the line width by latitude even within a single tile. That'd be cool if we could do it, but it's probably unrealistic. Setting the line width based on the average latitude or something would cause a discontinuity between adjacent tiles, so I don't think that would be good.